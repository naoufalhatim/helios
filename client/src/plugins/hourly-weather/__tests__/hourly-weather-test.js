import React from 'react';
import renderer from 'react-test-renderer';
import HourlyWeather from '../hourly-weather';
import moment from 'moment';

jest.mock('react-skycons', () => 'Skycons');

const testHours = [
  {time: moment().startOf('day').add(1, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(2, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(3, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(4, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(5, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(6, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(7, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(8, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'},
  {time: moment().startOf('day').add(9, 'hour').unix(), temperature: 70, icon: 'partly-cloudy', precipPercent: 20, precipType: 'rain'}
];

describe('HourlyWeather Component', () => {
  it('renders', () => {
    const hourlyOutput = renderer.create(
      <HourlyWeather hours={testHours} />
    );

    expect(hourlyOutput.toJSON()).toMatchSnapshot();
  });
});
