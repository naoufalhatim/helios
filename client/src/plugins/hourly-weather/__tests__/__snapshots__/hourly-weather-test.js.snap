exports[`HourlyWeather Component renders 1`] = `
<div
  className="hourly-forecast">
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      1am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      2am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      3am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      4am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      5am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      6am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      7am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
  <div
    className="hour">
    <h3
      className="label-secondary hour-details">
      8am
    </h3>
    <h2
      className="label-primary hour-title">
      70°
    </h2>
    <div
      className="forecast-skycon">
      <Skycons
        autoplay={true}
        color="#fff"
        icon="PARTLY_CLOUDY" />
    </div>
    <h3
      className="label-secondary">
      20% rain
    </h3>
  </div>
</div>
`;
